packages/augur-core/source/contracts/GnosisSafeRegistry.sol

```diff
@@ -40,46 +56,83 @@ contract GnosisSafeRegistry is Initializable {
         _cash.approve(_fillOrder, MAX_APPROVAL_AMOUNT);
         _shareToken.setApprovalForAll(_fillOrder, true);
 
-        GnosisSafeRegistry _gnosisSafeRegistry = GnosisSafeRegistry(_gnosisSafeRegistry);
-        _gnosisSafeRegistry.register();
-
         _affiliates.setFingerprint(_fingerprint);
 
         if (_referralAddress != address(0)) {
             _affiliates.setReferrer(_referralAddress);
         }
     }
 
-    function register() external {
-        IGnosisSafe _safe = getAndValidateSafeFromSender();
+    function proxyCreated(address _proxy, address _mastercopy, bytes calldata _initializer, uint256 _saltNonce) external {
+        require(msg.sender == address(proxyFactory));
+        IGnosisSafe _safe = IGnosisSafe(_proxy);
+        bytes32 _codeHash;
+        assembly {
+            _codeHash := extcodehash(_safe)
+        }
+        require(_codeHash == proxyCodeHash, "Safe instance does not match expected code hash of the Proxy contract");
+        require(_mastercopy == gnosisSafeMasterCopy, "Proxy master contract is not the Gnosis Safe");
+        require(_safe.getThreshold() == 1, "Safe may only have a threshold of 1");
         address[] memory _owners = _safe.getOwners();
+        require(_owners.length == 1, "Safe may only have 1 user");
+
+        validateSafeCreation(_owners, _proxy, _mastercopy, _initializer, _saltNonce);
+
         address _owner = _owners[0];
         accountSafes[_owner].push(_safe);
     }
 
+    function validateSafeCreation(address[] memory _owners, address _proxy, address _mastercopy, bytes memory _initializer, uint256 _saltNonce) internal {
+        IAffiliates _affilifates = IAffiliates(affiliates);
+        bytes32 _fingerprint = _affilifates.getAccountFingerprint(_proxy);
+        address _referralAddress = _affilifates.getReferrer(_proxy);
+        bytes memory _expectedSetupForAugurData = abi.encodeWithSelector(this.setupForAugur.selector,
+            address(augur),
+            createOrder,
+            fillOrder,
+            cash,
+            shareToken,
+            affiliates,
+            _fingerprint,
+            _referralAddress
+        );
+        address _to;
+        bytes memory _data;
+        assembly {
+            // Skip selector and length:
+            let _initializerData := add(_initializer, 36)
+            // Load the to and data params
+            _to := mload(add(_initializerData, 64))
+            _data := add(_initializerData, mload(add(_initializerData, 96)))
+        }
+        require(_to == address(this));
+        // Requires the expected and actual data arguments are equal
+        require(_data.length == _expectedSetupForAugurData.length && keccak256(_data) == keccak256(_expectedSetupForAugurData));
```

Comment:
keccak256 equality isn't enough?

Reply:
it _might_ be, but figured this would ensure there were no possible ways to fake equality